src/c++/perf_analyzer/inference_profiler.h

```diff
@@ -540,6 +536,8 @@ class InferenceProfiler {
 
 #ifndef DOCTEST_CONFIG_DISABLE
   friend TestInferenceProfiler;
+
+ protected:
```

Comment:
Why protected?

Reply:
because i couldnt get access to the constructor otherwise in a derived class.